src/connectors/hierarchical-menu/connectHierarchicalMenu.js

```diff
@@ -293,19 +290,59 @@ export default function connectHierarchicalMenu(renderFn, unmountFn = noop) {
       },
 
       getWidgetSearchParameters(searchParameters, { uiState }) {
-        if (
+        const values =
           uiState.hierarchicalMenu &&
-          uiState.hierarchicalMenu[hierarchicalFacetName]
-        ) {
-          return searchParameters
-            .clearRefinements(hierarchicalFacetName)
-            .toggleRefinement(
-              hierarchicalFacetName,
-              uiState.hierarchicalMenu[hierarchicalFacetName].join(separator)
-            );
-        } else {
-          return searchParameters;
+          uiState.hierarchicalMenu[hierarchicalFacetName];
+
+        if (searchParameters.isHierarchicalFacet(hierarchicalFacetName)) {
+          const facet = searchParameters.getHierarchicalFacetByName(
+            hierarchicalFacetName
+          );
+
+          warning(
```

Comment:
The behavior with the warning is a bit different. Previously we were returning an empty configuration which was merged with the previous one. Now we override the facet definition. I don't think it matters to have one or the other, but the current solution match with the expectation of the function to be driven by the UI state & self-contained.